src/crewai/crew.py

```diff
@@ -729,20 +714,18 @@ def _handle_conditional_task(
             return skipped_task_output
         return None
 
-    def _prepare_agent_tools(self, task: Task, manager: Optional[BaseAgent]):
+    def _prepare_agent_tools(self, task: Task):
```

Comment:
This is the function that is going to cause issues during async calls.

Let's say we have a crew with 4 tasks and the first 3 are async.

when we prepare the agent tools for each async function we are going to update the manager tools.

Since there is only one manager, we are going to overwrite what tools the manager can use in each async call which is not what we want.

I'm happy to brainstorm a solution with you on this.